allennlp/service/predictors/simple_tagger.py

```diff
@@ -8,15 +15,26 @@ class SimpleTaggerPredictor(Predictor):
     """
     Wrapper for the :class:`~allennlp.models.bidaf.SimpleTagger` model.
     """
-    def predict_json(self, inputs: JsonDict) -> JsonDict:
+    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+        super(SimpleTaggerPredictor, self).__init__(model, dataset_reader)
+        self._tokenizer = WordTokenizer()
+
+    @overrides
+    def _json_to_instance(self, json: JsonDict) -> Instance:
         """
         Expects JSON that looks like ``{"sentence": "..."}``
         and returns JSON that looks like
         ``{"tags": [...], "class_probabilities": [[...], ..., [...]]}``
         """
-        sentence = inputs["sentence"]
-
-        tokens = TextField(self.tokenizer.tokenize(sentence)[0],
-                           token_indexers=self.token_indexers)
+        sentence = json["sentence"]
+        tokens, _ = self._tokenizer.tokenize(sentence)
+        return self._dataset_reader.text_to_instance(tokens)
 
-        return sanitize(self.model.tag(tokens))
+    @overrides
+    def _predictions_to_json(self, model_outputs: Dict[str, Any]) -> JsonDict:
+        predictions = model_outputs['class_probabilities']
+        argmax_indices = numpy.argmax(predictions, axis=-1)
+        tags = [self._model.vocab.get_token_from_index(x, namespace="labels")
+                for x in argmax_indices]
+        model_outputs['tags'] = tags
+        return super(SimpleTaggerPredictor, self)._predictions_to_json(model_outputs)
```

Comment:
I would argue this is more complex than just having `return sanitize(model_outputs)` in each `Predictor`.